### PR TITLE
Updates to LBANN to support ROCm-enabled DistConv

### PR DESCRIPTION
**Pull request by benson31**
Depends on LLNL/DiHydrogen#65.

### CODE REVIEW

**Review by KIwabuchi — APPROVED** [review]
Looks good to me!

File: src/callbacks/check_gradients.cpp [hunk 1]
```diff
@@ -133,10 +133,13 @@ struct CheckWeightsFunctor : DefaultErrorReporter
     // Get weights matrix and gradient
     const auto& weights_matrix = dtw.get_values();
     const auto& gradient = dtw.get_optimizer()->get_gradient();
-
+    // std::cout << "*** CHECKING WEIGHTS: NAME=" << dtw.get_name() << ", SIZE="
```
Comment (bvanessen) [inline src/callbacks/check_gradients.cpp:136]: Eventually this debug code should be removed.

**Review by bvanessen — APPROVED** [review]
LGTM.  CI unit tests pass on Tioga.

File: src/callbacks/check_gradients.cpp (hunk 1 above)
Comment (benson31) [reply src/callbacks/check_gradients.cpp]: Done

**Pull request merged**